### PR TITLE
Implement load and write as async

### PR DESCRIPTION
**Pull request by VirxEC**
This adds a new feature (+ dependency) `async-std`.

When the feature is enabled, two additional functions are implemented on top of everything else: `load_async` and `write_async`.

This is primarily useful for me, in my project, as I'm loading many cfg files, and being able to run other code while a file is being read from the disk is very useful.

So here I am with this code and PR. It was straightforward and didn't take long, but this is the most popular parsing crate for this file type I could find.

### CODE REVIEW

**Review by QEDK — CHANGES_REQUESTED** [review]
LGTM, can you add the test-cases too, thanks!

**Comment by VirxEC** [comment]
Done, verified they work

**Comment by VirxEC** [comment]
Actually the async_write test doesn't work if the indexmap feature isn't enabled, give me a second

**Comment by VirxEC** [comment]
Done, combined the two tests into one as well

**Comment by QEDK** [comment]
@VirxEC sorry to bother again! could you rebase it against master, you might also want to just use https://docs.rs/async-std/latest/async_std/fs/fn.read_to_string.html for loading the string, it's probably faster than all the other matches I had added.

**Comment by VirxEC** [comment]
Pulled upstream changes and re-implemented load_async in what I think is the most optimized way. Hope I did it right haha.

File: src/ini.rs [hunk 1]
```diff
@@ -851,3 +854,34 @@ impl Ini {
...
+    ///
+    ///Usage is similar to `load`, but `.await` must be called after along with the usual async rules.
+    ///
+    ///Returns `Ok(map)` with a clone of the stored `Map` if no errors are thrown or else `Err(error_string)`.
+    ///Use `get_mut_map()` if you want a mutable reference.
+    pub async fn load_async<T: AsRef<AsyncPath>>(
+        &mut self,
+        path: T,
+    ) -> Result<Map<String, Map<String, Option<String>>>, String> {
+        let s = async_fs::read_to_string(&path)
```
Comment (QEDK) [inline src/ini.rs:871]: We don't need to assign the variable here, something like:
```rust
        self.map = match self.parse(match async_fs::read_to_string(&path).await {
            Err(why) => return Err(format!(
                    "couldn't read {}: {}",
                    &path.as_ref().display(),
                    why
                )),
            Ok(s) => s,
            }) {
                Err(why) => {
                    return Err(format!(
                        "couldn't read {}: {}",
                        &path.as_ref().display(),
                        why
                    ))
                }
                Ok(map) => map,
        };
```
I've benchmarked the synchronous version, it's a bit faster (~1 microsecond), async benching is a bit too much of a headache.

**Review by QEDK — CHANGES_REQUESTED** [review]

File: src/ini.rs (hunk 1 above)
Comment (VirxEC) [reply src/ini.rs:871]: Assigning to a variable is more readable, in my opinion. Especially when mapping error values. It's your project, though, I'll change it.

**Review by QEDK — APPROVED** [review]
LGTM!

**Comment by QEDK** [comment]
Thanks for the contribution @VirxEC 🥳

**Comment by VirxEC** [comment]
I was also thinking of re-implementing #28 as I personally tested it & the code works but it seems that the PR author has disappeared.

**Pull request merged**